src/trg-preferences-dialog.c

```diff
@@ -592,6 +626,73 @@ static void add_profile_cb(GtkWidget * w, gpointer data)
     gtk_combo_box_set_active_iter(combo, &iter);
 }
 
+#ifdef HAVE_LIBSECRET
+static void
+password_set_cb(GObject *source, GAsyncResult *result, gpointer userdata)
+{
+    GtkWidget *win = gtk_widget_get_toplevel(GTK_WIDGET(userdata));
+
+    if(!secret_password_store_finish (result, NULL)) {
+        GtkWidget *dialog = gtk_message_dialog_new(GTK_WINDOW(win),
+                                                   GTK_DIALOG_MODAL,
+                                                   GTK_MESSAGE_ERROR,
+                                                   GTK_BUTTONS_OK,
+                                                   "%s", _("Failed to save password to keychain"));
+
+        gtk_window_set_title(GTK_WINDOW(dialog), _("Error"));
+        gtk_dialog_run(GTK_DIALOG(dialog));
```

Comment:
Never use `gtk_dialog_run()`. Show the dialog and connect to the `response` signal.